community/kernel/src/main/java/org/neo4j/kernel/impl/store/counts/CountsSnapshot.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store.counts;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.neo4j.kernel.impl.store.counts.keys.CountsKey;
+
+public class CountsSnapshot
+{
+    public static final CountsSnapshot NO_SNAPSHOT = null;
+    final private Map<CountsKey,long[]> map;
+    final private long txId;
+
+    public CountsSnapshot( long txId, Map<CountsKey,long[]> map )
+    {
+        this.map = map;
+        this.txId = txId;
+    }
+
+    public CountsSnapshot( long txId )
+    {
+        this( txId, new ConcurrentHashMap<>() );
```

Comment:
All other maps in this class are not concurrent. Why this one is? Imo if this class should be thread-safe than `Map` should be `ConcurrentMap` everywhere.